packages/minimongo/minimongo.js

```diff
@@ -696,12 +696,33 @@ LocalCollection.prototype.update = function (selector, mod, options, callback) {
   // it. (We don't need to save the original results of paused queries because
   // they already have a resultsSnapshot and we won't be diffing in
   // _recomputeResults.)
-  var qidToOriginalResults = {};
+  var qidToOriginalResults = {},
+      docMap = {}, // We should only clone each document once, even if it appears in multiple queries
+      onlyIDQuery = _.size(selector) === 1 && _.isString(selector._id); // If the selector is just looking up a single _id, then we only need to clone that document
```

Comment:
The `_id` field can also be an integer or a MongoID. There's a `LocalCollection._selectorIsIdPerhapsAsObject` utility function that should do the trick. And that would also make this code correctly trigger when `selector` is an ID and not an object with an `_id` field.